nupic/network/synapse.py

```diff
@@ -0,0 +1,38 @@
+﻿# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2014, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+
+class Synapse(object):

```

Comment:
Not sure how fast your fingers work :-) You might have answered this before I asked the follow up question about the Synapse in the actual code? 

I wonder what your example would do for an object accessed by:
```Python
self._synapses[synapse].permanence = xx.xx
```

rather than a dispatched method call like setdefault() in the above examples? I bet with 10M (as opposed to 20M) calls and assignments like the above it would be faster than create?